reflect_test.go

```diff
@@ -167,4 +167,7 @@ func TestNameConversion(t *testing.T) {
 	schema := reflector.Reflect(&SimpleStruct{})
 	_, exists := schema.Definitions["SimpleStruct"].Properties.Get("simpleproperty")
 	require.True(t, exists)
+
+	schemaJSON, _ := json.Marshal(schema)
+	require.Contains(t, string(schemaJSON), "simpleproperty")
```

Comment:
Sorry, I don't think I conveyed my concern... this is testing the schema, we want to test the serialisation of the struct itself... to ensure the schema matches the struct (I don't believe it will).

Reply:
Ah, I see what you're saying. Is it really possible to ensure they match though? 

The serialization of the struct and the creation of the schema are done by different packages so I don't see an easy/nice way of syncing both of them up.

Reply:
Yeah that's exactly my point... jsonschema will use `json:"<name>"` tags for the names, so it will currently be correct. With this change, the schema will say one thing but the JSON encoded objects will have (potentially) different names.

Reply:
What I had in mind was to use this in conjunction with [spf13/viper](https://github.com/spf13/viper) which is case-insensitive, so I don't necessarily see the the mismatch here being a problem especially as an optional hook. Thoughts?